src/test/java/com/chikli/hudson/plugin/naginator/NaginatorPublisherTest.java

```diff
@@ -0,0 +1,148 @@
+import hudson.model.AbstractBuild;
+import hudson.model.BuildListener;
+import hudson.tasks.Builder;
+
+/**
+ * Tests for {@link NaginatorPublisher}.
+ * Many tests for {@link NaginatorPublisher} are also in {@link NaginatorListenerTest}
+ */
+public class NaginatorPublisherTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    
+    private static class FailSpecificAxisBuilder extends Builder {
+        private final String combinationFilterToFail;
+        
+        public FailSpecificAxisBuilder(String combinationFilterToFail) {
+            this.combinationFilterToFail = combinationFilterToFail;
+        }
+        
+        @Override
+        public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
+                throws InterruptedException, IOException {
+            if (!(build instanceof MatrixRun)) {
+                throw new AbortException("only applicable for MatrixRun");
+            }
+            
+            MatrixRun run = (MatrixRun)build;
+            
+            if (run.getParent().getCombination().evalGroovyExpression(
+                    run.getParent().getParent().getAxes(),
+                    combinationFilterToFail
+            )) {
+                throw new AbortException("Combination matches the filter.");
+            }
+            
+            return true;
+        }
+    }
+    
+    @Test
+    public void testMatrixBuild() throws Exception {
+        // all children are rescheduled if rerunMatrixPart is not set.
+        {
+            MatrixProject p = j.createMatrixProject();
+            AxisList axes = new AxisList(
+                    new Axis("axis1", "1", "2"),
+                    new Axis("axis2", "1", "2")
+            );
+            p.setAxes(axes);
+            p.getBuildersList().add(new FailSpecificAxisBuilder("(axis1=='1' && axis2=='2') || (axis1=='2' && axis2=='1')"));
+            p.getPublishersList().add(new NaginatorPublisher(
+                    null,   // regexpForRerun
+                    false,  // rerunIfUnstable
+                    false,  // rerunMatrixPart
+                    false,  // checkRegexp
+                    1,      // maxSchedule
+                    new FixedDelay(0)
+            ));
+            
+            p.scheduleBuild2(0);
+            j.waitUntilNoActivity();
+            
+            // retries 1 time
+            assertEquals(2, p.getLastBuild().number);
+            
+            // (1, 1), (1, 2), (2, 1), (2, 2) are rescheduled.
+            MatrixBuild b = p.getLastBuild();
+            assertNotNull(b.getExactRun(new Combination(axes, "1", "1")));
+            assertNotNull(b.getExactRun(new Combination(axes, "1", "2")));
+            assertNotNull(b.getExactRun(new Combination(axes, "2", "1")));
+            assertNotNull(b.getExactRun(new Combination(axes, "2", "2")));
+        }
+        
+        // only failed children are rescheduled if rerunMatrixPart is set.
+        {
```

Comment:
* To save execution time of tests as Jenkins is launched for each tests.
* Those blocks both are to test the behavior for multi-configuration tests.

They are not so important and I can split it into 2 tests.
Do you like me splitting also tests in `NaginatorScheduleActionTest`?